### PR TITLE
Notify users about denied access on ask-to-join  rooms

### PR DESCRIPTION
**Pull request by nurjinjafar**
We https://github.com/nordeck are currently implementing the knock rooms behind the feature flag feature_ask_to_join (introduced in https://github.com/matrix-org/matrix-react-sdk/pull/11182).

This pull request introduces the deny mask for requests.

Epic: https://github.com/vector-im/element-web/issues/18655

Relates to
https://github.com/matrix-org/matrix-react-sdk/pull/11404
https://github.com/matrix-org/matrix-js-sdk/pull/3673
https://github.com/matrix-org/matrix-react-sdk/pull/11353
https://github.com/matrix-org/matrix-js-sdk/pull/3647
https://github.com/matrix-org/matrix-react-sdk/pull/11248
https://github.com/matrix-org/matrix-react-sdk/pull/11182
https://github.com/vector-im/element-web/pull/25923

Screens

Screenshot 2023-08-29 at 14 15 48

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

Here's what your changelog entry will look like:
Type: enhancement
 
✨ Features
Introduce Deny access request mask on Ask-to-join rooms (https://github.com/matrix-org/matrix-react-sdk/pull/11480). Contributed by @nurjinjafar.

---
Here's what your changelog entry will look like:

## ✨ Features
 * Notify users about denied access on ask-to-join  rooms ([\#11480](https://github.com/matrix-org/matrix-react-sdk/pull/11480)). Contributed by @nurjinjafar.

### CODE REVIEW

**Comment by charlynguyen** [comment]
Relates to https://github.com/vector-im/element-web/issues/26084

**Comment by nurjinjafar** [comment]
Hello @dbkr , Hi @kerryarchibald, could you please give us an update regarding the pull request in case any exist?

File: src/components/views/rooms/RoomPreviewBar.tsx [hunk 1]
```diff
@@ -188,7 +189,11 @@ export default class RoomPreviewBar extends React.Component<IProps, IState> {
         const myMember = this.getMyMember();
 
         if (myMember) {
+            const previousMembership = myMember.events.member?.event?.unsigned?.prev_content?.membership;
```
Comment (kerryarchibald) [inline src/components/views/rooms/RoomPreviewBar.tsx:192]: `myMember.events.member?.getPrevContent().membership`

File: test/components/views/rooms/RoomPreviewBar-test.tsx [hunk 2]
```diff
@@ -45,12 +45,14 @@ const makeMockRoomMember = ({
...
     memberContent,
+    oldMembership,
 }: {
     userId?: string;
     isKicked?: boolean;
-    membership?: "invite" | "ban";
+    membership?: "invite" | "ban" | "leave";
     content?: Partial<IContent>;
     memberContent?: Partial<IContent>;
+    oldMembership?: "Join" | "knock";
```
Comment (kerryarchibald) [inline test/components/views/rooms/RoomPreviewBar-test.tsx:55]: should be lowercase `"join"`

File: src/components/views/rooms/RoomPreviewBar.tsx [hunk 3]
```diff
@@ -397,6 +402,21 @@ export default class RoomPreviewBar extends React.Component<IProps, IState> {
                 }
                 break;
             }
+            case MessageCase.RequestDenied: {
+                title = _t("You have been denied access");
+
+                subTitle = _t(
+                    "As you have been denied access, please wait for an invitation from a room admin or moderator.",
```
Comment (kerryarchibald) [inline src/components/views/rooms/RoomPreviewBar.tsx:409]: The [designs on the linked ticket](https://github.com/vector-im/element-web/issues/18655#issuecomment-1570432810) use different copy - is there a reason this is different?

**Review by kerryarchibald — COMMENTED** [review]
Code looks good! I have left a couple of small nitpicks.

File: src/components/views/rooms/RoomPreviewBar.tsx (hunk 3 above)
Comment (nurjinjafar) [reply src/components/views/rooms/RoomPreviewBar.tsx]: you are right! the ticket on our side had different text! but i added the text from the design!

**Review by kerryarchibald — APPROVED** [review]

**Pull request merged**